src/lib/widget/_utilities/component.utilities.ts

```diff
@@ -15,52 +16,67 @@ export interface ComponentStoreValue {
 }
 
 export const componentStore: Writable<ComponentStoreValue> = writable({
-  modal: null,
-  form: null,
+  lastAction: null,
   error: null,
+  form: null,
+  modal: null,
   mounted: false,
   open: null,
   reason: null,
   type: null,
 });
 
-export const componentApi = () => {
-  const { update, subscribe } = componentStore;
+/**
+ * @function closeComponent - this is a widget internal function not to be exposed to user
+ * @param {object} args - object containing  the reason for closing component
+ * @param {boolean} shouldCloseDialog - if true, the close command comes from outside of dialog component
+ */
+export function closeComponent(
+  args?: { reason: ComponentStoreValue['reason'] },
+  shouldCloseDialog?: boolean,
+) {
+  componentStore.update((state) => {
+    if (state.type === 'inline') {
+      console.warn('Component type of "inline" has no `close` method');
+      // There's nothing to do, so just return existing state
+      return state;
+    }
+    if (!state.modal?.component) {
+      console.warn('Modal component is not mounted. Please instantiate the Widget before use.');
+      // There's nothing to do, so just return existing state
+      return state;
+    }
```

Comment:
This is a private API to be used internally. This helps address an infinite loop issue from closing the modal.